packages/openjpeg.rb

```diff
@@ -1,41 +1,36 @@
-require 'package'
+require 'buildsystems/cmake'
 
-class Openjpeg < Package
+class Openjpeg < CMake
   description 'An open source JPEG 2000 codec, written in C.'
   homepage 'https://github.com/uclouvain/openjpeg'
-  version '2.5.0'
+  version '2.5.0-1'
   license 'BSD-2'
-  compatibility 'all'
+  compatibility 'x86_64 aarch64 armv7l'
   source_url 'https://github.com/uclouvain/openjpeg/archive/v2.5.0.tar.gz'
   source_sha256 '0333806d6adecc6f7a91243b2b839ff4d2053823634d4f6ed7a59bc87409122a'
 
   binary_url({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0_armv7l/openjpeg-2.5.0-chromeos-armv7l.tar.zst',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0_armv7l/openjpeg-2.5.0-chromeos-armv7l.tar.zst',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0_i686/openjpeg-2.5.0-chromeos-i686.tar.zst',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0_x86_64/openjpeg-2.5.0-chromeos-x86_64.tar.zst'
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0-1_armv7l/openjpeg-2.5.0-1-chromeos-armv7l.tar.zst',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0-1_armv7l/openjpeg-2.5.0-1-chromeos-armv7l.tar.zst',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openjpeg/2.5.0-1_x86_64/openjpeg-2.5.0-1-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
-    aarch64: 'e8f47ab6ee2e585d41a40def616a99e8c38855cb286a687c264bddc62666f41b',
-     armv7l: 'e8f47ab6ee2e585d41a40def616a99e8c38855cb286a687c264bddc62666f41b',
-       i686: '3cea296254991ddd62be39c5ec3cc901343ec4e06cc5896146780b14fb270cd6',
-     x86_64: '47261041b12e3bb63bc2cd47d285afcc8fabdcd1075b80db0c1a35bf3267a826'
+    aarch64: 'a7af46fa74fd908fafa8c9ba3761a5f3167159e7e0776d72892b342d130db604',
+     armv7l: 'a7af46fa74fd908fafa8c9ba3761a5f3167159e7e0776d72892b342d130db604',
+     x86_64: 'cad4bc6d4f0784d9f0871c51d36c5c96ff5becff4d43ed52eefda13c89ea5eeb'
   })
 
-  def self.build
-    Dir.mkdir 'builddir'
-    Dir.chdir 'builddir' do
-      system "cmake \
-        -G Ninja \
-        #{CREW_CMAKE_OPTIONS} \
-        -DBUILD_SHARED_LIBS=ON \
-        -DOPENJPEG_INSTALL_LIB_DIR='lib#{CREW_LIB_SUFFIX}' \
-        .."
-    end
-    system 'ninja -C builddir'
-  end
+  depends_on 'glibc' # R
+  depends_on 'jbigkit' # R
+  depends_on 'libdeflate' # R
+  depends_on 'libjpeg' # R
+  depends_on 'libpng' # R
+  depends_on 'libtiff' # R
+  depends_on 'libwebp' # R
+  depends_on 'xzutils' # R
+  depends_on 'zlibpkg' # R
+  depends_on 'zstd' # R
 
-  def self.install
-    system "DESTDIR=#{CREW_DEST_DIR} ninja -C builddir install"
-  end
+  cmake_options "-DBUILD_SHARED_LIBS=ON \
```

Comment:
`DBUILD_SHARED_LIBS` is `ON` by default in `2.5.0`.

https://github.com/uclouvain/openjpeg/blob/v2.5.0/CMakeLists.txt#L186